### PR TITLE
refactor kfp-api's on_remove to use KRH built-in delete

### PR DESCRIPTION
**Pull request by ca-scribner**
Refactors kfp-api charm's `on_remove()` event handler to use KubernetesResourceHandler.delete() rather than its own removal procedure.  This was done to:
* standardise on a removal procedure with other charms
* remove the need for the `on_remove()` handler to have a KRH object with full `context`.  In future we will add another Kubernetes object here that requires relation data to render, and this relation data may not be available during the removal event

### CODE REVIEW

File: charms/kfp-api/src/charm.py [hunk 1]
```diff
@@ -554,14 +571,20 @@ def _on_upgrade(self, _):
...
         self.unit.status = MaintenanceStatus("Removing K8S resources")
-        k8s_resources_manifests = self.k8s_resource_handler.render_manifests()
-        try:
-            delete_many(self.k8s_resource_handler.lightkube_client, k8s_resources_manifests)
-        except ApiError as error:
-            # do not log/report when resources were not found
-            if error.status.code != 404:
-                self.logger.error(f"Failed to delete K8S resources, with error: {error}")
-                raise error
+        # Define the KubernetesResourceHandler manually because the object storage confict in
```
Comment (ca-scribner) [inline charms/kfp-api/src/charm.py:574]:
```suggestion
        # Define the KubernetesResourceHandler manually because the object storage conflict in
```

**Comment by misohu** [comment]
LGTM I see kfp-api integration failing because of mysql error fixed with this https://github.com/canonical/kfp-operators/pull/386 just the bundle test is failing on `test_create_and_monitor_recurring_run`. Maybe we can try to rerun. If it occurs again lets open issue and we can merge this change.

**Comment by ca-scribner** [comment]
We will rebase and retarget this on main after #362 is merged

**Review by NohaIhab — APPROVED** [review]
LGTM

**Pull request merged**